File: conductor/learning_hub_sdk/examples/quick_start.py
"""
Quick Start Example for Learning Hub SDK
"""

from learning_hub_sdk import LearningHubClient, AuthenticationError

def main():
    """Quick start example."""
    # Initialize client (use API key or username/password)
    try:
        client = LearningHubClient(
            base_url="https://api.learninghub.com",
            api_key="your-api-key-here"
            # OR
            # username="your-username",
            # password="your-password"
        )
        
        # Health check
        print("Checking API health...")
        health = client.health_check()
        print(f"API Status: {health}")
        
        # Get current user
        print("
Getting current user...")
        user = client.get_current_user()
        print(f"User: {user.username} ({user.email})")
        
        # List courses
        print("
Listing courses...")
        courses = client.get_courses(page_size=5)
        print(f"Found {len(courses.results)} courses:")
        for course in courses.results:
            print(f"  - {course.title} (${course.price})")
        
        # Get specific course
        if courses.results:
            course_id = courses.results[0].id
            print(f"
Getting course details for ID {course_id}...")
            course = client.get_course(course_id)
            print(f"Course: {course.title}")
            print(f"Description: {course.description}")
            print(f"Duration: {course.duration_hours} hours")
        
        # Create enrollment
        if courses.results and user:
            print(f"
Enrolling in course...")
            enrollment = client.enroll_user(user.id, courses.results[0].id)
            print(f"Enrolled at: {enrollment.enrolled_at}")
        
    except AuthenticationError as e:
        print(f"Authentication failed: {e}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    main()

File: conductor/learning-hub-js-sdk/examples/quick-start.ts
/**
 * Quick Start Example for Learning Hub JavaScript/TypeScript SDK
 */

import { LearningHubClient, AuthenticationError } from '../src/index';

async function main() {
  try {
    // Initialize client (use API key or username/password)
    const client = new LearningHubClient({
      baseURL: 'https://api.learninghub.com',
      apiKey: 'your-api-key-here'
      // OR
      // username: 'your-username',
      // password: 'your-password'
    });

    // Health check
    console.log('Checking API health...');
    const health = await client.healthCheck();
    console.log('API Status:', health);

    // Get current user
    console.log('\nGetting current user...');
    const user = await client.getCurrentUser();
    console.log(`User: ${user.fullName} (${user.email})`);

    // List courses
    console.log('\nListing courses...');
    const courses = await client.getCourses({ page_size: 5 });
    console.log(`Found ${courses.count} courses:`);
    courses.results.forEach(course => {
      console.log(`  - ${course.title} (${course.formattedPrice})`);
    });

    // Get specific course
    if (courses.results.length > 0) {
      const course = courses.results[0];
      console.log(`\nGetting course details for ID ${course.id}...`);
      console.log(`Course: ${course.title}`);
      console.log(`Description: ${course.description}`);
      console.log(`Duration: ${course.formattedDuration}`);

      // Create enrollment
      console.log(`\nEnrolling in course...`);
      const enrollment = await client.enrollUser(user.id, course.id);
      console.log(`Enrolled at: ${enrollment.enrolled_at}`);

      // Get progress
      const progress = await client.getUserProgress(user.id, course.id);
      console.log(`Progress: ${progress.formattedProgress}`);
    }

  } catch (error) {
    if (error instanceof AuthenticationError) {
      console.error(`Authentication failed: ${error.message}`);
    } else {
      console.error(`Error: ${error}`);
    }
  }
}

main();
